checker.c

```diff
@@ -1,21 +1,40 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+int is_temp_ok(float temp) {
+    return (temp >= 0 && temp <= 45);
+}
+
+int is_soc_ok(float soc) {
+    return (soc >= 20 && soc <= 80);
+}
+
+int is_charge_rate_ok(float chargeRate) {
+    return (chargeRate <= 0.8);
+}
+
+//function to check overall battery Health
+int is_battery_ok(float temp, float soc, float chargeRate) {
+    if (is_temp_ok(temp) && is_soc_ok(soc) && is_charge_rate_ok(chargeRate))
+    {
+        return 1;
+    } else {
+        return 0;
+    }
 }
 
 int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+    //test case 1
+    if (is_battery_ok(20,70,0.7)) {
```

Comment:
main is not responsible for printing status of Battery! wrap printf in another function